upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go

```diff
@@ -208,6 +208,10 @@ func (t *LaunchTemplate) Find(c *fi.Context) (*LaunchTemplate, error) {
 			actual.SecurityGroups = append(actual.SecurityGroups, &SecurityGroup{ID: id})
 		}
 	}
+	// In older Kops versions, security groups were added to LaunchTemplateData.SecurityGroupIds
+	for _, id := range lt.LaunchTemplateData.SecurityGroupIds {
+		actual.SecurityGroups = append(actual.SecurityGroups, &SecurityGroup{ID: fi.String("legacy-" + *id)})
```

Comment:
Why the "legacy-" prefix?
I thought we just want to move the SGs from `LaunchTemplateData.SecurityGroupIds` to `LaunchTemplateData.NetworkInterfaces[0].Groups`.

Reply:
`Find()` builds a representation of the "actual" object, the state as it currently exists. The caller then diffs that against the representation of the "expected" object, the state that was built from the config in the state store. Only if those two differ does it call `RenderAWS`, which would build a new launch template to attach to the ASG.

If we did not add the "legacy-" prefix then the actual could equal the expected and thus it would believe that no changes were necessary.

Reply:
In this particular case, if SG is in the legacy place, I think the LT should not change. Only on next change the SG will be moved together with something else to the new location.

I guess it is debatable if we want this or not. Your choice here.